projects/reservoir/train_ks.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+import dacite
+import logging
+import numpy as np
+from sklearn.linear_model import Ridge
+from sklearn.utils import shuffle
+
+import yaml
+from fv3fit.reservoir import (
+    ReservoirComputingReadout,
+    Reservoir,
+    ReservoirComputingModel,
+)
+from fv3fit.reservoir.config import ReservoirTrainingConfig
+from ks import KSConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "ks_config", type=str, help=("Config file for KS equation parameters")
+    )
+    parser.add_argument("train_config", type=str, help=("Config file for training"))
+    return parser
+
+
+def add_input_noise(arr, stddev):
+    return np.random.normal(loc=0, scale=stddev, size=arr.shape)
+
+
+def transform_inputs_to_reservoir_states(X, reservoir, input_noise: float):
+    reservoir_states = [
+        reservoir.state,
+    ]
+    X_noised = add_input_noise(arr=X, stddev=input_noise)
+    for x in X_noised:
+        reservoir.increment_state(x)
+        reservoir_states.append(reservoir.state)
+    return np.array(reservoir_states)
+
+
+if __name__ == "__main__":
+    parser = _get_parser()
+    args = parser.parse_args()
+    with open(args.ks_config, "r") as f:
+        ks_config = dacite.from_dict(KSConfig, yaml.safe_load(f))
+    with open(args.train_config, "r") as f:
+        train_config_dict = yaml.safe_load(f)
+        train_config = ReservoirTrainingConfig.from_dict(train_config_dict)
+
+    training_ts = ks_config.generate(
+        n_steps=train_config.n_samples + train_config.n_burn, seed=train_config.seed
+    )
+    training_ts_burnin, training_ts_keep = (
+        training_ts[: train_config.n_burn],
+        training_ts[train_config.n_burn :],
+    )
+
+    reservoir = Reservoir(train_config.reservoir_hyperparameters)
+
+    reservoir.synchronize(
+        add_input_noise(arr=training_ts_burnin, stddev=train_config.input_noise)
+    )
+    training_reservoir_states = transform_inputs_to_reservoir_states(
+        X=training_ts_keep, reservoir=reservoir, input_noise=train_config.input_noise
+    )
+
+    # first training reservoir state is the reservoir state after last burn in step
+    # first training target state is the system state after last burn in step
+    X_train = training_reservoir_states[:-1]
```

Comment:
I think it could be more readable if the low-level detail of trimming off the final reservoir state happened in `training_reservoir_states` rather than here.